src/Common/src/System/IO/ChunkedMemoryStream.cs

```diff
@@ -0,0 +1,118 @@
+// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
I like how you were able to refactor this common code into something that can be re-used within all of CoreFx.